tests/suites/test_suite_pkcs5.function

```diff
@@ -14,12 +14,14 @@ void pbkdf2_hmac( int hash, data_t * pw_str, data_t * salt_str,
 {
     unsigned char key[100];
 
+    PSA_INIT();
```

Comment:
This is OK but I think we could be even more precise: define a new macro `PSA_INIT_IF_NO_MD()` that only calls `psa_crypto_init()` if `MD_C` is not defined. That way, we can check that things still work without `psa_crypto_init()` when `MD_C` is defined, as the documentation say. Wdyt?

Reply:
Right, that's a good thing to test. Do you think that it will be useful in other .function files, or just for pkcs5?

Reply:
It seems likely to be used in other files. Actually, even in this PR, it's already used in pkparse too IIRC.

Reply:
Code that only uses hashes should not require `psa_crypto_init`, which among other things requires entropy.

Reply:
> Code that only uses hashes should not require `psa_crypto_init`, which among other things requires entropy.

PBKDF2 uses HMAC, which in PSA requires the key store to be initialized unless I missed something.

Even it was just hashing, I don't think there's any point in the documentation right now that says you can use `psa_hash_xxx()` functions without calling `psa_crypto_init()` first, and I'd rather not rely on undocumented properties. (Also, isn't there some kind of driver initialization step that needs to happen when hashes are provided by drivers?)

So, if you're talking about the future, with split init and the parts that can be done transparently happening automatically under the hood, then yes, sure, but as far as this PR is concerned, I think right now we do need to call `psa_crypto_init()`.